src/services/infra/DynamoDBService.ts

```diff
@@ -1,82 +1,50 @@
-import {
-  DeleteCommandOutput,
-  GetCommandOutput,
-  UpdateCommandOutput,
-} from "@aws-sdk/lib-dynamodb"
+import { AttributeValue } from "@aws-sdk/client-dynamodb"
+import { DeleteCommandOutput } from "@aws-sdk/lib-dynamodb"
 import autoBind from "auto-bind"
 
 import { config } from "@config/config"
 
-import { SiteLaunchMessage } from "@root/../microservices/site-launch/shared/types"
+import {
+  SiteLaunchMessage,
+  isSiteLaunchMessage,
+} from "@root/../microservices/site-launch/shared/types"
 
-import DynamoDBClient, { UpdateParams } from "./DynamoDBClient"
+import DynamoDBClient from "./DynamoDBClient"
 
-const MOCK_LAUNCH: SiteLaunchMessage = {
-  repoName: "my-repo",
-  appId: "my-app",
-  primaryDomainSource: "example.com",
-  primaryDomainTarget: "myapp.example.com",
-  domainValidationSource: "example.com",
-  domainValidationTarget: "myapp.example.com",
-  requestorEmail: "john@example.com",
-  agencyEmail: "agency@example.com",
-  githubRedirectionUrl: "https://github.com/my-repo",
-  redirectionDomain: [
-    {
-      source: "example.com",
-      target: "myapp.example.com",
-      type: "A",
-    },
-  ],
-  status: { state: "pending", message: "PENDING_DURING_SITE_LAUNCH" },
-}
 export default class DynamoDBService {
   private readonly dynamoDBClient: DynamoDBClient
 
   private readonly TABLE_NAME: string
 
-  constructor() {
-    this.dynamoDBClient = new DynamoDBClient()
+  constructor(dynamoDBClient: DynamoDBClient) {
+    this.dynamoDBClient = dynamoDBClient
     this.TABLE_NAME = config.get("aws.dynamodb.siteLaunchTableName")
```

Comment:
should this be part of the arguments to the constructor so that we can send in mock values if required for testing?

Reply:
good point! 5af08260e793cc6874eb1d7728178d47648b7b54